internal/cluster/transition.go

```diff
@@ -524,10 +525,50 @@ func (th *transitionHandler) PostPreparingTimedOut(sw stateswitch.StateSwitch, a
 		updatedCluster *common.Cluster
 	)
 
+	// Capture some information about hosts states, we should be able to determine more about the cause of the timeout.
+	// This can occur because the host...
+	//
+	// 1: Becomes disconnected - the "connected" validation of the host will be failing
+	// 2: The media has been disconnected - The "media connected" validation of the host will be failing
+	// 3: The condition "installation-disk-speed-check-succesful" is unsuccessful
+	// 4: The condition "succesful-container-image-availability" is unsuccessful
+	//
+	issues := []string{}
+
+	// Iterate over cluster hosts and check to see if there are any problematic validations
+	// if so, report on these by setting "installation_preparation_completion_status", "installation_preparation_completion_status_reason"
+	for _, clusterHost := range sCluster.cluster.Hosts {
+		if clusterHost.ValidationsInfo != "" {
+			// Deserialize validations info
+			var hostValidations ValidationsStatus
+			err = json.Unmarshal([]byte(clusterHost.ValidationsInfo), &hostValidations)
+			if err != nil {
+				issues = append(issues, fmt.Sprintf("unable to deserialize validations info for host %s due to error %s", clusterHost.ID, err.Error()))
+			}
+			for _, vRes := range hostValidations {
+				for _, v := range vRes {
+					// Any unsuccesful validation should be noted as these are likely to have caused issues
+					if v.Status != ValidationSuccess {
```

Comment:
I'm not sure if this part is correct, in the timeout that i saw the result of the validation was success but didn't run.
If i remember correctly the validation between the condition and the validation info is different, the validation marked as success with the info that state that it didn't run yet, but the condition is failing.